array_test.go

```diff
@@ -1582,6 +1582,35 @@ func TestArray_Transform(t *testing.T) {
 
 		newArray.chain.assertFailed(t)
 	})
+
+	t.Run("canonization", func(t *testing.T) {
+		type (
+			myInt int
+		)
+
+		reporter := newMockReporter(t)
+		array := NewArray(reporter, []interface{}{2, 4, 6})
+
+		newArray := array.Transform(func(_ int, val interface{}) interface{} {
+			if val, ok := val.(float64); ok {
+				return myInt(val)
+			}
+			t.Errorf("failed transformation")
+			return nil
+		})
+
+		assert.Equal(t, []interface{}{float64(2), float64(4), float64(6)}, newArray.Raw())
+		newArray.chain.assertNotFailed(t)
+		newArray.chain.clearFailed()
+
+		assert.NotEqual(t, []interface{}{"2.0", float64(4), float64(6)}, newArray.Raw())
+		newArray.chain.assertNotFailed(t)
+		newArray.chain.clearFailed()
+
+		assert.Equal(t, []interface{}{2.0, float64(4), 6.0}, newArray.Raw())
+		newArray.chain.assertNotFailed(t)
+		newArray.chain.clearFailed()
```

Comment:
These three checks can be reduced to only one:

```go
		assert.Equal(t, []interface{}{2.0, 4.0, 6.0}, newArray.Raw())
		newArray.chain.assertNotFailed(t)
```

Reply:
done